packages/mui-base/src/BackdropUnstyled/BackdropUnstyled.js

```diff
@@ -2,57 +2,38 @@ import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
 import composeClasses from '../composeClasses';
-import isHostComponent from '../utils/isHostComponent';
-import { getBackdropUtilityClass } from './backdropUnstyledClasses';
-
-const useUtilityClasses = (ownerState) => {
-  const { classes, invisible } = ownerState;
+import appendOwnerState from '../utils/appendOwnerState';
+import { getBackdropUnstyledUtilityClass } from './backdropUnstyledClasses';
 
+const useUtilityClasses = () => {
   const slots = {
-    root: ['root', invisible && 'invisible'],
+    root: ['root'],
   };
 
-  return composeClasses(slots, getBackdropUtilityClass, classes);
+  return composeClasses(slots, getBackdropUnstyledUtilityClass, {});
 };
 
 const BackdropUnstyled = React.forwardRef(function BackdropUnstyled(props, ref) {
-  const {
-    classes: classesProp,
-    className,
-    invisible = false,
-    component = 'div',
-    components = {},
-    componentsProps = {},
-    /* eslint-disable react/prop-types */
-    theme,
-    ...other
-  } = props;
+  const { className, component, components = {}, componentsProps = {}, ...other } = props;
 
-  const ownerState = {
-    ...props,
-    classes: classesProp,
-    invisible,
-  };
+  const ownerState = props;
 
   const classes = useUtilityClasses(ownerState);
 
-  const Root = components.Root || component;
-  const rootProps = componentsProps.root || {};
-
-  return (
-    <Root
-      aria-hidden
-      {...rootProps}
-      {...(!isHostComponent(Root) && {
-        as: component,
-        ownerState: { ...ownerState, ...rootProps.ownerState },
-        theme,
-      })}
-      ref={ref}
-      {...other}
-      className={clsx(classes.root, rootProps.className, className)}
-    />
+  const Root = component ?? components.Root ?? 'div';
+  const rootProps = appendOwnerState(
+    Root,
+    {
+      'aria-hidden': true,
```

Comment:
Looks like this is the only thing that makes this component useful. Is having the component too much? Should we add in the docs that the component used as a backdrop should have an `aria-hidden`: true? In the modal for example, we can set this automatically. 

Just a thought.. :)

Reply:
Alright, I will close the PRs related to it and update the demos where necessary.